src/verifier.cc

```diff
@@ -268,22 +453,30 @@ Result Verifier::Probe(const ProbeCommand* command,
   }
 
   if (count_of_invalid_pixels) {
-    const uint8_t* p = ptr + row_stride * (first_invalid_j + y) +
-                       texel_stride * (x + first_invalid_i);
+    const auto& component = texel_format->GetComponents().back();
+    float scale_factor_for_error_report = 1.0f;
+    if (component.mode == FormatMode::kUNorm ||
+        component.mode == FormatMode::kSNorm ||
+        component.mode == FormatMode::kSRGB) {
+      scale_factor_for_error_report = 255.0f;
+    }
+
     return Result(
         "Line " + std::to_string(command->GetLine()) +
         ": Probe failed at: " + std::to_string(x + first_invalid_i) + ", " +
-        std::to_string(first_invalid_j + y) + "\n" +
-        "  Expected RGBA: " + std::to_string(command->GetR() * 255) + ", " +
-        std::to_string(command->GetG() * 255) + ", " +
-        std::to_string(command->GetB() * 255) +
-        (command->IsRGBA() ? ", " + std::to_string(command->GetA() * 255) +
+        std::to_string(first_invalid_j + y) + "\n" + "  Expected RGBA: " +
+        std::to_string(command->GetR() * scale_factor_for_error_report) + ", " +
+        std::to_string(command->GetG() * scale_factor_for_error_report) + ", " +
+        std::to_string(command->GetB() * scale_factor_for_error_report) +
+        (command->IsRGBA() ? ", " +
+                                 std::to_string(command->GetA() *
+                                                scale_factor_for_error_report) +
                                  "\n  Actual RGBA: "
                            : "\n  Actual RGB: ") +
-        std::to_string(static_cast<int>(p[0])) + ", " +
-        std::to_string(static_cast<int>(p[1])) + ", " +
-        std::to_string(static_cast<int>(p[2])) +
-        (command->IsRGBA() ? ", " + std::to_string(static_cast<int>(p[3]))
+        std::to_string(actual_color_texel_.r) + ", " +
```

Comment:
Why do we need actual_color_texel_ can't we just copy the failing vector of values in the loop above? No need for the class value.

Reply:
done